platform/country-converter/platform-country-converter-api/src/main/java/org/codice/ddf/country/converter/api/CountryCodeConverter.java

```diff
@@ -0,0 +1,63 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.country.converter.api;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Service to convert country codes into various formats. In order to convert country codes from
+ * format X into format Y and vice versa, a properties file containing country code mappings is
+ * required. Since ISO 3166-1 alpha 3 is the normalized format, all country code mappings file
+ * should be formatted from format X into ISO 3166-1 alpha 3, and not from format X directly into
+ * format Y. In order to convert from format X into format Y, two properties files are required,
+ * one for format X and one for format Y. The file path of the file also needs to be configured via
+ * the feature's configuration. The name of the properties file (excluding the .extension) will be
+ * used as the value of the "sourceFormat" or "desiredFormat" method parameters for the convert
+ * method. To specify the normalized format, use the predefined string "ISO_3166_1_ALPHA_3".
+ * <p>
+ * Example:
+ * When converting from FIPS 10-4 to ISO 3166-1 alpha 3, if the name of the country code mappings
+ * file is: "fips_10_4.properties", the string "FIPS_10_4" (case-insensitive) is used as the value
+ * of the source country code format and "ISO_3166_1_ALPHA3" as the value of the desired country
+ * code format.
+ * <p>
+ * For more information on how the country code mappings are formatted, refer to the
+ * Administrator's documentation.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface CountryCodeConverter {
+
+    /**
+     * Variable representing the normalized country code format.
+     */
+    String ISO_3166_1_ALPHA_3 = "ISO_3166_1_ALPHA3";
+
+    /**
+     * Converts a country code from one format into another. If the country code maps to more than
+     * one country code from the desired format, all of them will be returned in a list of strings.
+     * The string value of the country code formats used for "sourceFormat" and "desiredFormat" must
+     * be equal to the name of their corresponding country code mappings file, excluding the file
+     * extension. EX. file name: "fips_10_4.properties", format used for conversion: "fips_10_4"
+     *
+     * @param countryCode   a country code.
+     * @param sourceFormat  source format of the country code.
+     * @param desiredFormat desired format of the country code to convert into.
+     * @return country code(s) in a List of Strings.
+     */
+    List<String> convert(@Nullable String countryCode, String sourceFormat, String desiredFormat);
```

Comment:
Sorry, looks like my comment wasn't clear :-(

My suggestion wasn't really the opposite of Phil's solution. I was basically proposing to use the source and destination formats for looking up the service instead of passing those arguments to the method itself.